### PR TITLE
Enhance Datumaro data format stream importer performance

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary
 - Ticket no. 120785
 - Change streaming import logic with DatumPageMapper implemented in Rust

| Before | After |
| :-: | :-: |
| image | image |

### How to test
Our existing tests cover this change.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [x] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/utils/test_utils.py [hunk 1]
```diff
@@ -45,15 +44,20 @@ def __enter__(self):
...
+                except unittest.SkipTest:
+                    # Suppress skip test errors from git.util.rmtree
+                    if not exc_type:
+                        raise
+            else:
+                rmfile(self.path)
+
+            if not os.path.exists(self.path):
+                return
+            sleep(0.5)
```
Comment (wonjuleee) [inline tests/utils/test_utils.py:60]: why is this required?

**Review by wonjuleee — COMMENTED** [review]
In overall, it looks good to me, but could you change the target branch to release/1.5.0?

File: tests/utils/test_utils.py (hunk 1 above)
Comment (vinnamkim) [reply tests/utils/test_utils.py:60]: If the permission error occurs in L50, it will retry requesting removal at most 10 times.

**Comment by vinnamkim** [comment]
> In overall, it looks good to me, but could you change the target branch to release/1.5.0?

Changed it to `releases/1.5.0`

**Review by wonjuleee — APPROVED** [review]

**Comment by vinnamkim** [comment]
These are final performance comparison results between `StreamDataset` and `Dataset` (non-`StreamDataset`) for COCO, Datum, VOC, and YOLO data formats:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=ko style='font-family:Calibri;font-size:11.0pt'>


<div style='direction:ltr'>


  | StreamDataset | Dataset
-- | -- | --
COCO | image | image 
Datumaro | image |  image
VOC | image | image 
YOLO | image | image

</div>


</body>

</html>
